x/smart/stream_dialer.go

```diff
@@ -507,9 +545,43 @@ func (f *StrategyFinder) NewDialer(ctx context.Context, testDomains []string, co
 		testDomains[di] = makeFullyQualified(domain)
 	}
 
-	dialer, err := f.newProxylessDialer(ctx, testDomains, parsedConfig)
-	if err != nil && parsedConfig.Fallback != nil {
-		return f.findFallback(ctx, testDomains, parsedConfig.Fallback)
+	// Fast resume the winning strategy from the cache
+	if f.Cache != nil {
+		rankedConfig, first2Try := f.rankStrategiesFromCache(inputConfig)
+		if first2Try != nil {
+			if dialer, _, err := f.findFallback(ctx, testDomains, first2Try); err == nil {
```

Comment:
Let's return a single element above.
```suggestion
			if dialer, _, err := f.findFallback(ctx, testDomains, [first2Try]); err == nil {
```